tests/api/test_variant.py

```diff
@@ -558,3 +563,314 @@ def test_fetch_unpublished_variant_anonymous_user(
+    assert attribut_value_count == size_attribute.values.count()
+
+
+def test_product_variant_bulk_create_with_new_attribute_value(
+    staff_api_client, product, size_attribute, permission_manage_products
+):
+    product_varinat_count = ProductVariant.objects.count()
+    attribut_value_count = size_attribute.values.count()
+    product_id = graphene.Node.to_global_id("Product", product.pk)
+    attribut_value = size_attribute.values.last()
+    variants = [
+        {
+            "sku": str(uuid4())[:12],
+            "attributes": [
+                {"slug": size_attribute.slug, "values": [attribut_value.name]}
+            ],
+        },
+        {
+            "sku": str(uuid4())[:12],
+            "attributes": [{"slug": size_attribute.slug, "values": ["Test-attribute"]}],
+        },
+    ]
+
+    variables = {"productId": product_id, "variants": variants}
+    staff_api_client.user.user_permissions.add(permission_manage_products)
+    response = staff_api_client.post_graphql(
+        PRODUCT_VARIANT_BULK_CREATE_MUTATION, variables
+    )
+    content = get_graphql_content(response)
+    data = content["data"]["productVariantBulkCreate"]
+    assert not data["bulkProductErrors"]
+    assert data["count"] == 2
+    assert product_varinat_count + 2 == ProductVariant.objects.count()
+    assert attribut_value_count + 1 == size_attribute.values.count()
+
+
+def test_product_variant_bulk_create_negative_quantity(
+    staff_api_client, product, size_attribute, permission_manage_products
+):
+    product_varinat_count = ProductVariant.objects.count()
+    product_id = graphene.Node.to_global_id("Product", product.pk)
+    variants = [
+        {
+            "sku": str(uuid4())[:12],
+            "quantity": -1000,
+            "attributes": [{"slug": size_attribute.slug, "values": ["Test-value"]}],
+        },
+        {
+            "sku": str(uuid4())[:12],
+            "quantity": 100,
+            "attributes": [{"slug": size_attribute.slug, "values": ["Test-value2"]}],
+        },
+    ]
+
+    variables = {"productId": product_id, "variants": variants}
+    staff_api_client.user.user_permissions.add(permission_manage_products)
+    response = staff_api_client.post_graphql(
+        PRODUCT_VARIANT_BULK_CREATE_MUTATION, variables
+    )
+    content = get_graphql_content(response)
+    data = content["data"]["productVariantBulkCreate"]
+    assert len(data["bulkProductErrors"]) == 1
+    error = data["bulkProductErrors"][0]
+    assert error["field"] == "quantity"
+    assert error["code"] == ProductErrorCode.INVALID.name
+    assert error["index"] == 0
+    assert product_varinat_count == ProductVariant.objects.count()
+
+
+def test_product_variant_bulk_create_duplicated_sku(
+    staff_api_client, product, size_attribute, permission_manage_products
+):
+    product_varinat_count = ProductVariant.objects.count()
+    product_id = graphene.Node.to_global_id("Product", product.pk)
+    sku = product.variants.first().sku
```

Comment:
Can we ensure that if the SKU is from another product, it will return the error as well?